graphs.hpp

```diff
@@ -869,14 +866,20 @@ namespace graphs
 			for (size_t y = ay >= ymax ? 0 : yaxis - (ay / ystep); y < yaxis and y < height; ++y)
 				aaarray[x][y] = acolor;
 		}
-
-		aoptions.type = type_histogram;
+	
+		if (aoptions.type != type_histogram) {
+			options aoptions_hist = aoptions;
+			aoptions_hist.type = type_histogram;
+		}
+		else {
+			return graph(height, width, xmin, xmax, ymin, ymax, aaarray, aoptions);
+		}
```

Comment:
The `aoptions_hist` variable is unused. Maybe you meant something like this:
```suggestion
		if (aoptions.type != type_histogram)
		{
			options hist_options = aoptions;
			hist_options.type = type_histogram;
			return graph(height, width, xmin, xmax, ymin, ymax, aaarray, hist_options);
		}
```

Reply:
Good catch! I originally created the new options object outside the if scope and changed my approach halfway, forgot to follow through